File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumTipoPagamentoAutorizacaoPagamento.js
let EnumTipoPagamentoAutorizacaoPagamentoHelper = function () {
    this.Todas = 0;
    this.PagamentoAdiantamento = 1;
    this.PagamentoSaldo = 2;
};

EnumTipoPagamentoAutorizacaoPagamentoHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Pagamento Adiantamento", value: this.PagamentoAdiantamento },
            { text: "Pagamento Saldo", value: this.PagamentoSaldo }
        ];
    },
    obterOpcoesPesquisa: function () {
        return [{ text: "Todas", value: this.Todas }].concat(this.obterOpcoes());
    }
};

let EnumTipoPagamentoAutorizacaoPagamento = Object.freeze(new EnumTipoPagamentoAutorizacaoPagamentoHelper());

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumTipoPagamentoAutorizacaoPagamento.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum EnumTipoPagamentoAutorizacaoPagamento
    {
        PagamentoAdiantamento = 1,
        PagamentoSaldo = 2
    }

    public static class EnumTipoPagamentoAutorizacaoPagamentoHelper
    {
        public static string ObterDescricao(this EnumTipoPagamentoAutorizacaoPagamento tipoPagamento)
        {
            switch (tipoPagamento)
            {
                case EnumTipoPagamentoAutorizacaoPagamento.PagamentoAdiantamento: return "Pagamento Adiantamento";
                case EnumTipoPagamentoAutorizacaoPagamento.PagamentoSaldo: return "Pagamento Saldo";
                default: return string.Empty;
            }
        }

    }
}
